### PR TITLE
Feat/gh3 connect hub controller; fixes #3 

### LINKED ISSUE #3: Connect HubController to establish connection with a 10-25 source

### PR DESCRIPTION
**Pull request by nathanatrhea**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/DEHP-ECOSIMPRO/pulls) open
- [x] I have verified that I am following the DEHP-EcosimPro [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/DEHP-ECOSIMPRO/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description
fixes #3 

- [x] Connected the HubController
- [x] Opening Session via the login view

### CODE REVIEW

File: DEHPEcosimPro/App.xaml.cs [hunk 1]
```diff
@@ -46,7 +53,29 @@ public App(ContainerBuilder containerBuilder = null)
         {
             var splashScreenViewModel = new DXSplashScreenViewModel() { Title = "DEHP-EcosimPro Adapter"};
             SplashScreenManager.Create(() => new SplashScreen(), splashScreenViewModel).ShowOnStartup();
+            containerBuilder ??= new ContainerBuilder();
+            RegisterViewModels(containerBuilder);
             AppContainer.BuildContainer(containerBuilder);
         }
+
+        protected override void OnStartup(StartupEventArgs e)
```
Comment (lxatrhea) [inline DEHPEcosimPro/App.xaml.cs:61]: missing docs

File: DEHPEcosimPro/ViewModel/DataSourceViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        /// </summary>
+        private readonly IHubController hubController;
+
+        /// <summary>
+        /// Backing field for <see cref="ConnectButtonText"/>
+        /// </summary>
+        private string connectButtonText = ConnectText;
+
+        /// <summary>
+        /// Gets or sets the name
```
Comment (lxatrhea) [inline DEHPEcosimPro/ViewModel/DataSourceViewModel.cs:69]: A bit short on the docs

File: DEHPEcosimPro/ViewModel/DataSourceViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        }
+        
+        /// <summary>
+        /// <see cref="ReactiveCommand{T}"/> for connecting to a data source
+        /// </summary>
+        public ReactiveCommand<object> ConnectCommand { get; set; }
+
+        /// <summary>
+        /// Initializes a new <see cref="DataSourceViewModel"/>
+        /// </summary>
```
Comment (lxatrhea) [inline DEHPEcosimPro/ViewModel/DataSourceViewModel.cs:84]: missing param docs

File: DEHPEcosimPro/ViewModel/Interfaces/IDataSourceViewModel.cs [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+{
+    using ReactiveUI;
+
+    /// <summary>
+    /// Definition of methods and properties of <see cref="DataSourceViewModel"/>
+    /// </summary>
+    public interface IDataSourceViewModel
+    {
+        /// <summary>
+        /// Gets or sets the name
```
Comment (lxatrhea) [inline DEHPEcosimPro/ViewModel/Interfaces/IDataSourceViewModel.cs:35]: Name is a bit generic in docs

**Review by lxatrhea — APPROVED** [review]
Looks good, apart from some docs issues.

File: DEHPEcosimPro/ViewModel/Interfaces/IMainWindowViewModel.cs [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+{
+    /// <summary>
+    /// Interface definitions of methods and properties of <see cref="Views.MainWindow"/>
+    /// </summary>
+    public interface IMainWindowViewModel
+    {
+        /// <summary>
+        /// Gets the view model that represents the 10-25 data source
+        /// </summary>
+        IDataSourceViewModel Object1025 { get; }
```
Comment (alexatrhea) [inline DEHPEcosimPro/ViewModel/Interfaces/IMainWindowViewModel.cs:35]: Use better names for these Properties like `HubDataSourceViewModel`

File: DEHPEcosimPro/ViewModel/Interfaces/IMainWindowViewModel.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+        /// <summary>
+        /// Gets the view model that represents the EcosimPro data source
+        /// </summary>
+        IDataSourceViewModel ObjectEcosimPro { get; }
```
Comment (alexatrhea) [inline DEHPEcosimPro/ViewModel/Interfaces/IMainWindowViewModel.cs:40]: Same here

**Review by alexatrhea — APPROVED** [review]

**Pull request merged**